### ISSUE TITLE
Server locale detector does not work with .yml translation

### ISSUE THREAD

**Issue by Applelo** [nuxt-modules/i18n#2835]
### Environment

- Operating System: `Darwin`
- Node Version:     `v20.11.1`
- Nuxt Version:     `3.10.3`
- CLI Version:      `3.10.1`
- Nitro Version:    `2.9.3`
- Package Manager:  `pnpm@8.15.4`
- Builder:          `-`
- User Config:      `modules`, `security`, `htmlValidator`, `css`, `imports`, `multiCache`, `graphqlMiddleware`, `runtimeConfig`, `i18n`, `piniaPersistedstate`, `experimental`, `image`, `prismic`, `vite`
- Runtime Modules:  `@nuxtjs/i18n@8.1.1`
- Build Modules:    `-`

### Reproduction

YML (not working): https://stackblitz.com/edit/nuxt-i18n-localedector-yml?file=nuxt.config.ts 
JSON (working): https://stackblitz.com/edit/nuxt-i18n-localedector-yml-nuxna8?file=nuxt.config.ts

### Describe the bug

When we are using the experimental `localeDetector` option with `.yml` file translation and they are a problem with the compilation. There is no problem when we are using json file.

### Logs

```shell
ERROR  [worker reload] [worker init] _import_meta_url_ is not defined                 
  at .nuxt/dev/index.mjs:1:65
  at ModuleJob.run (node:internal/modules/esm/module_job:218:25)
  at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
  at async loadESM (node:internal/process/esm_loader:28:7)
  at async handleMainPromise (node:internal/modules/run_main:113:12)
```

**Comment by BobbieGoede** [comment]
I'm getting the same error in the playground after updating `nitropack` to `2.9.0` or higher, maybe related https://github.com/nuxt/nuxt/issues/26170.

I haven't checked further (yet) if this is related to `.yml` files as you described.

**Comment by Applelo** [comment]
I have added the a new reproduction (forked from the previous one) and change all `.yml` file into `.json` and it is working.

**Comment by BobbieGoede** [comment]
Now that https://github.com/unjs/nitro/pull/2275 has been merged this issue can be resolved by overriding `nitropack` to install its nightly release (you will have to remove lockfiles before reinstalling). I have updated you reproduction to demonstrate [here](https://stackblitz.com/edit/nuxt-i18n-localedector-yml-2rtfks?file=package.json).

Not sure when then next nitro release will be, let's hope it's soon and that Nuxt includes it in the next release!

Let's keep this issue open until that's released, until then if anyone else runs into this add this to your `package.json` as a temporary workaround:

```json
  "overrides": {
    "nitropack": "npm:nitropack-nightly"
  }
```
